# Lecture 3 Examples and Case — Python Basics

**Course:** AAU E26 — Introduction to Scripting, Data Mining and Machine Learning  
**Lecture:** Lecture 3  
**Goal:** Run short examples about values, variables, types, errors, and one service record.

[Open this notebook in Google Colab](https://colab.research.google.com/github/asmrabbi/E26_TAN7_Scripting_CPH/blob/main/notebooks/examples/L03_examples_python_basics.ipynb) · [View the course repository](https://github.com/asmrabbi/E26_TAN7_Scripting_CPH)

Run the cells from top to bottom. Every executable line includes a short comment explaining what it does.


## Goal

Learn how Python stores values, calculates results, converts types, reports errors, and describes one service record.


## Setup

No extra packages or files are required for this notebook.


## Steps

### 1. Print a message


In [1]:
course_message = "Welcome to Python in Google Colab!"  # Stores a greeting as text in a named variable.
print(course_message)  # Displays the value stored in the variable.


Welcome to Python in Google Colab!


### 2. Store values and inspect their types


In [2]:
service_name = "Housing support"  # Stores a text value, which Python calls a string.
cases_received = 120  # Stores a whole number, which Python calls an integer.
average_days = 5.1  # Stores a decimal number, which Python calls a float.
target_met = True  # Stores a True-or-False value, which Python calls a Boolean.
print(type(service_name))  # Displays the Python type of the service name.
print(type(cases_received))  # Displays the Python type of the case count.
print(type(average_days))  # Displays the Python type of the average.
print(type(target_met))  # Displays the Python type of the target flag.


<class 'str'>
<class 'int'>
<class 'float'>
<class 'bool'>


### 3. Calculate a rate and build a readable sentence


In [3]:
cases_resolved = 112  # Stores the number of cases resolved during the month.
resolution_rate = cases_resolved / cases_received  # Divides resolved cases by received cases to create a proportion.
resolution_percentage = resolution_rate * 100  # Converts the proportion into a percentage.
summary = f"Resolved {resolution_percentage:.1f}% of received cases."  # Formats the percentage with one decimal place.
print(summary)  # Displays the completed summary sentence.


Resolved 93.3% of received cases.


### 4. Convert text into a number


In [4]:
score_text = "4.2"  # Simulates a number that arrived from a form as text.
score_number = float(score_text)  # Converts the text into a decimal number that Python can calculate with.
updated_score = score_number + 0.1  # Adds a small amount after the conversion succeeds.
print(updated_score)  # Displays the calculated result.


4.3


### 5. Simulate user input without pausing the lecture


In [5]:
typed_city = "Copenhagen"  # Simulates the value a user might type into an input box.
clean_city = typed_city.strip().title()  # Removes surrounding spaces and applies consistent capitalisation.
print(f"City recorded as: {clean_city}")  # Displays the cleaned city value.


City recorded as: Copenhagen


### 6. Read and repair a common error safely


In [6]:
incorrect_value = "five"  # Stores text that cannot be converted directly into a number.
try:  # Starts a protected block so the demonstration does not stop the notebook.
    numeric_value = float(incorrect_value)  # Attempts the conversion that will raise a ValueError.
except ValueError as error_message:  # Catches the expected error and gives it a readable name.
    print(type(error_message).__name__)  # Displays the type of error that Python reported.
    print("Repair: replace the word with digits such as 5.0.")  # Explains one appropriate repair in plain language.


ValueError
Repair: replace the word with digits such as 5.0.


### 7. Recurring case: describe one monthly service record


In [7]:
record_id = 1001  # Stores a unique identifier for the service record.
report_month = "2026-01"  # Stores the month represented by the record.
city = "Copenhagen"  # Stores the city connected to the observation.
service_type = "Housing"  # Stores the category of service.
cases_received = 120  # Stores the number of new cases.
cases_resolved = 112  # Stores the number of completed cases.
unresolved_cases = cases_received - cases_resolved  # Calculates how many received cases were not resolved.
print(f"Record {record_id}: {service_type} in {city}, {report_month}.")  # Displays the identifying fields.
print(f"Unresolved cases: {unresolved_cases}")  # Displays the calculated difference.


Record 1001: Housing in Copenhagen, 2026-01.
Unresolved cases: 8


## Checks

Expected highlights: `str`, `int`, `float`, and `bool` appear as types; the resolution rate is about 93.3%; the protected conversion reports `ValueError`; unresolved cases equal 8.


## Next Steps

Open the matching student exercise notebook and change one value at a time before rerunning the relevant cell.
